backend/infrahub/core/schema/definitions/core/action.py

```diff
@@ -0,0 +1,260 @@
+    attributes=[
+        Attr(
+            name="action_type",
+            kind="Text",
+            enum=GroupAction.available_types(),
+            optional=False,
+            order_weight=3000,
+        ),
+    ],
+    relationships=[
+        Rel(
+            name="group",
+            peer=InfrahubKind.GENERICGROUP,
+            kind=RelKind.ATTRIBUTE,
+            cardinality=Cardinality.ONE,
+            identifier="core_action__group",
+            optional=False,
+            order_weight=4000,
+        ),
+    ],
+)
+
+
+core_node_trigger_rule = NodeSchema(
+    name="NodeTriggerRule",
+    namespace="Core",
+    include_in_menu=True,
+    icon="mdi:filter-cog-outline",
+    label="Node Trigger Rule",
+    default_filter="name__value",
+    order_by=["name__value"],
+    display_labels=["name__value"],
+    branch=BranchSupportType.AGNOSTIC,
+    uniqueness_constraints=[["name__value"]],
+    generate_profile=False,
+    inherit_from=[InfrahubKind.TRIGGERRULE],
+    attributes=[
+        Attr(
+            name="node_kind",
+            kind="Text",
+            unique=False,
+            optional=False,
+            order_weight=3000,
+        ),
+        Attr(
+            name="mutation_action",
+            kind="Text",
+            enum=NodeAction.available_types(),
+            unique=False,
+            optional=False,
+            order_weight=4000,
+        ),
+        Attr(
+            name="schema_path",
+            kind="Text",
+            unique=False,
+            optional=False,
+            order_weight=5000,
+        ),
+        Attr(
+            name="value",
+            kind="Text",
+            unique=False,
+            optional=False,
+            order_weight=6000,
+        ),
+        Attr(
+            name="value_updated_to",
+            kind="Boolean",
+            unique=False,
+            optional=False,
+            default_value=True,
+            order_weight=7000,
+        ),
+        Attr(
+            name="triggering_branch",
+            kind="Text",
+            optional=False,
+            default_value=BranchMatch.DEFAULT_BRANCH.value,
+            enum=BranchMatch.available_types(),
+            order_weight=8000,
+        ),
+    ],
+    relationships=[],
+)
+
+core_group_trigger_rule = NodeSchema(
+    name="GroupTriggerRule",
+    namespace="Core",
+    include_in_menu=True,
+    icon="mdi:filter-cog-outline",
+    label="Group Trigger Rule",
+    default_filter="name__value",
+    order_by=["name__value"],
+    display_labels=["name__value"],
+    branch=BranchSupportType.AGNOSTIC,
+    uniqueness_constraints=[["name__value"]],
+    generate_profile=False,
+    inherit_from=[InfrahubKind.TRIGGERRULE],
+    attributes=[
+        Attr(
+            name="members_added",
+            kind="Boolean",
+            unique=False,
+            optional=False,
+            default_value=True,
+            order_weight=3000,
+        ),
+        Attr(
```

Comment:
it looks like all nodes inheriting from `TRIGGERRULE` have `triggering_branch`, if so should we move it to the generic instead ?

Reply:
The reason I had it at the node level was for future trigger rules where we might not have a branch. I.e. if we look at branch creation events or such. The alternative would be to ignore that attribute on those type of nodes, or always trigger alternatively raise a violation error within the mutation if someone tries to create such a trigger.

Reply:
I moved it to the generic instead, even though it might be a bit cleaner to have it on each "child node", it also means more replication. So it might just be better to ignore the setting if we have it on any node where we don't need it.